File: lib/som/java/src/neuralize/me/Distribution.java
package neuralize.me;

import java.util.ArrayList;
import java.util.List;

public class Distribution {
	
	public static List<Double> distribute(double startValue, double endValue, int size){
		List<Double> array = new ArrayList<Double>();
		double difference = endValue - startValue;
		double n = Math.max(size-1, 1);
		
		for(int i=0; i<size; i++)
			array.add(  startValue + i * difference / n );

		return array;
	}
	
	public static List<Integer> distributeInteger(double startValue, double endValue, int size){
		List<Double> doubleArray = distribute(startValue, endValue, size);
		List<Integer> array = new ArrayList<Integer>();

		for(Double value:doubleArray)
			array.add( value.intValue() );

		return array;
	}
	
}

File: lib/som/ruby/lib/distribution.rb
class RubySOM
  module Distribution
    
    def distribute(start_value, end_value, array_size, want_ints)
      diff = 1.0 * (end_value - start_value)
      n = [array_size-1, 1].max
      
      (0..(array_size-1)).map do |i|
          v = start_value + i * diff / n
          want_ints ? v.round : v
      end
    end
    
  end
end
